Objective-C/Tests/LogTest.m

```diff
@@ -301,58 +254,46 @@ - (void) testEnableAndDisableCustomLogging {
-    CBLDatabase.log.file.level = kCBLLogLevelDebug;
-    AssertEqual(CBLDatabase.log.file.config.maxSize, 1024);
-    AssertEqual(CBLDatabase.log.file.config.maxRotateCount, 2);
+    CBLLogSinks.file = [[CBLFileLogSink alloc] initWithLevel: kCBLLogLevelInfo directory: logFileDirectory];
+    AssertEqual(CBLLogSinks.file.maxFileSize, (NSInteger)kCBLDefaultFileLogSinkMaxSize);
+    AssertEqual(CBLLogSinks.file.maxKeptFiles, (NSUInteger)kCBLDefaultFileLogSinkMaxKeptFiles);
+    AssertEqual(CBLLogSinks.file.usePlaintext, kCBLDefaultLogFileUsePlaintext);
+    CBLLogSinks.file = [[CBLFileLogSink alloc] initWithLevel: kCBLLogLevelDebug directory: logFileDirectory usePlaintext: YES maxKeptFiles: 2 maxFileSize: 1024];
+    AssertEqual(CBLLogSinks.file.maxFileSize, 1024);
+    AssertEqual(CBLLogSinks.file.maxKeptFiles, 2);
     
     // this should create three files, as the 1KB + 1KB + extra ~400-500Bytes.
     [self writeOneKiloByteOfLog];
     [self writeOneKiloByteOfLog];
     
-    NSUInteger totalFilesShouldBeInDirectory = (CBLDatabase.log.file.config.maxRotateCount + 1) * 5;
+    NSUInteger totalFilesShouldBeInDirectory = CBLLogSinks.file.maxKeptFiles * 5;
 #if !DEBUG
     totalFilesShouldBeInDirectory = totalFilesShouldBeInDirectory - 1;
 #endif
-    NSArray* files = [self getLogsInDirectory: config.directory properties: nil onlyInfoLogs: NO];
+    NSArray* files = [self getLogsInDirectory: CBLLogSinks.file.directory properties: nil onlyInfoLogs: NO];
     AssertEqual(files.count, totalFilesShouldBeInDirectory);
 }
 
-- (void) testFileLoggingDisableLogging {
-    CBLLogFileConfiguration* config = [self logFileConfig];
-    config.usePlainText = YES;
-    CBLDatabase.log.file.config = config;
-    CBLDatabase.log.file.level = kCBLLogLevelNone;
-    
+- (void) testFileLoggingDisabled {
+    CBLLogSinks.file = nil;
+
     NSString* inputString = [[NSUUID UUID] UUIDString];
     [self writeAllLogs: inputString];
     
-    AssertFalse([self isKeywordPresentInAnyLog: inputString path: config.directory]);
+    AssertFalse([self isKeywordPresentInAnyLog: inputString path: CBLLogSinks.file.directory]);
 }
 
 - (void) testFileLoggingReEnableLogging {
-    CBLLogFileConfiguration* config = [self logFileConfig];
-    config.usePlainText = YES;
-    CBLDatabase.log.file.config = config;
-    CBLDatabase.log.file.level = kCBLLogLevelNone;
+    CBLLogSinks.file = nil;
     
     NSString* inputString = [[NSUUID UUID] UUIDString];
     [self writeAllLogs: inputString];
     
-    AssertFalse([self isKeywordPresentInAnyLog: inputString path: config.directory]);
+    AssertFalse([self isKeywordPresentInAnyLog: inputString path: CBLLogSinks.file.directory]);
     
-    CBLDatabase.log.file.level = kCBLLogLevelVerbose;
+    CBLLogSinks.file = [[CBLFileLogSink alloc] initWithLevel: kCBLLogLevelVerbose directory: logFileDirectory usePlaintext: YES maxKeptFiles: kCBLDefaultFileLogSinkMaxKeptFiles maxFileSize: kCBLDefaultLogFileMaxSize];
```

Comment:
Break it down to multiple lines. Generally, please keep the line up to around 120 chars.